tests/integration/integration_steps/cdf_steps.py

```diff
@@ -284,3 +289,38 @@ def assert_state_store_in_cdf(
 
             assert row is not None
             assert row.columns["high"] is not None
+
+
+def delete_event_state_store_in_cdf(
+    database: str, table: str, event_state_key: str, cognite_client: CogniteClient
+):
+    row = cognite_client.raw.rows.retrieve(database, table, event_state_key)
+    if row is not None:
+        cognite_client.raw.rows.delete(database, table, event_state_key)
+
+
+def push_events_to_cdf(cognite_client: CogniteClient, events: List[EventWrite]):
+    res = cognite_client.events.create(events)
+    assert confirm_events_in_cdf(
+        cognite_client, events
+    )  # Ensure all events are in CDF list operation before continuing test
+    return res
+
+
+def confirm_events_in_cdf(
+    cognite_client: CogniteClient, events: List[EventWrite], retries: int = 5
+):
+    event_external_ids = [event.external_id for event in events]
+    for _ in range(retries):
+        event_list = cognite_client.events.list(limit=None)
+        if set(event.external_id for event in event_list) == set(
+            event_external_ids
+        ):  # check if all events are in CDF - exact match
+            return True
+        sleep(2**_)   # wait before the next retry using exponential backoff
```

Comment:
Might be good to log the failure and the retry to the console.

Reply:
Added logging in debug mode